src/main/java/WebServer.java

```diff
@@ -0,0 +1,37 @@
+import java.io.*;
+
+public class WebServer {
+
+    private SocketManager serverSocketManager;
+    private RequestHandler requestHandler;
+    private ResponseHandler responseHandler;
+    private static final String RESPONSE_STATUS_CODE_200 = "HTTP/1.1 200 OK\n";
+    private static final String SIMPLE_GET_URL = "simple_get";
+
+    public WebServer(SocketManager serverSocketManager, RequestHandler requestHandler, ResponseHandler responseHandler) {
+        this.serverSocketManager = serverSocketManager;
+        this.requestHandler = requestHandler;
+        this.responseHandler = responseHandler;
+    }
+
+    public void start(int port) throws IOException {
+        serverSocketManager.connect(port);
+        prepareResponse();
+    }
+
+    private CommunicationChannel communicationChannel() {
+        return serverSocketManager.communicationChannel();
+    }
+
+    private void prepareResponse() throws IOException {
+        OutputStream output = communicationChannel().getOutput();
+       if (request().contains(SIMPLE_GET_URL)) {
+           responseHandler.prepare(output, RESPONSE_STATUS_CODE_200);
+       }
+        output.close();
+    }
+
+    private String request() throws IOException {
+       return requestHandler.read(communicationChannel().getInput());
```

Comment:
Although it is hidden by the fact that you call a private method of this class first, what is really happening is serverSocketManager.communicationChannel().getInput(). Are you happy with this?

Reply:
Ideally it would be `communicationChannel().getInput()` and `communicationChannel().getOutput()` but I need the `Socket` to create these channels which is created in the `serverSocketManager`. Unless `connect()` returns a `Socket` and I create the `communicationChannel` in `WebServer`?